File: Freestylecoding.Math.FSharp.Tests/Natural/EquatableNatural.fs
namespace Natural

open Xunit
open Freestylecoding.Math

type public EquatableNatural() =
    [<Theory>]
    [<InlineData(  0u,  0u, true)>]
    [<InlineData(  1u,  0u, false)>]
    [<InlineData(  0u,  1u, false)>]
    [<InlineData(  1u,  1u, true)>]
    member public this.Sanity left right expected =
        Assert.Equal(
            expected,
            Natural( [left] ).Equals( Natural( [right] ) )
        )

    [<Fact>]
    member public this.LargeNaturalsTrue () =
        Assert.True(
            Natural( [0xFu; 0x00000101u] ).Equals( Natural( [0xFu; 0x00000101u] ) )
        )

    [<Fact>]
    member public this.LargeNaturalsFalse () =
        Assert.False(
            Natural( [0x8u; 0x00000101u] ).Equals( Natural( [0xFu; 0x00000101u] ) )
        )

File: Freestylecoding.Math.CSharp.Tests/Natural/EquatableNatural.cs
using System;
using Xunit;

namespace Natural.Tests;

using static Freestylecoding.Math.CSharp.Tests.Helpers;
using Natural = Freestylecoding.Math.Natural;

public class EquatableNatural {
    [Theory]
    [InlineData( 0u, 0u, true )]
    [InlineData( 1u, 0u, false )]
    [InlineData( 0u, 1u, false )]
    [InlineData( 1u, 1u, true )]
    public void Sanity( uint left, uint right, bool expected ) =>
        Assert.Equal(
            expected,
            ( new Natural( left ) as IEquatable<Natural> ).Equals( new Natural( right ) )
        );

    [Fact]
    public void LargeNaturalsTrue() =>
        Assert.True(
            ( FListNatural( 0xFu, 0x00000101u ) as IEquatable<Natural> ).Equals( FListNatural( 0xFu, 0x00000101u ) )
        );

    [Fact]
    public void LargeNaturalsFalse() =>
        Assert.False(
            ( FListNatural( 0x8u, 0x00000101u ) as IEquatable<Natural> ).Equals( FListNatural( 0xFu, 0x00000101u ) )
        );
}